### PR TITLE
PLFM-7719: ETL to process node snapshots

### PR DESCRIPTION
**Pull request by marcomarasca**

### CODE REVIEW

File: src/scripts/glue_jobs/node_snapshots.py [hunk 1]
```diff
@@ -25,37 +24,55 @@ def get_dynamic_frame(connection_type, file_format, source_path, glue_context):
...
-# process the access record
 def transform(dynamic_record):
-    date = datetime.utcfromtimestamp(dynamic_record["change_timestamp"] / 1000.0)
+    date = datetime.utcfromtimestamp(dynamic_record["changeTimestamp"] / 1000.0)
+    
+    # Add the partition date from the timestamp of the change
     dynamic_record["change_date"] = date.strftime("%Y-%m-%d")
+    
+    # The records come in with the syn prefix, we need to remove that
+    dynamic_record["snapshot"]["id"] = strip_syn_prefix(dynamic_record["snapshot"]["id"])
```
Comment (thomasyu888) [inline src/scripts/glue_jobs/node_snapshots.py]: Just out of curiosity, why do syn ids need to be removed?

Comment (marcomarasca) [reply src/scripts/glue_jobs/node_snapshots.py]: The idea is that it's much better to have a bigint type in the queriable table (simpler to write queries, joins etc)

Comment (thomasyu888) [reply src/scripts/glue_jobs/node_snapshots.py]: Makes sense from a compute perspective, maybe a little more confusing from a user perspective (depending on who the audience is), but not too difficult to remove "syn"

File: src/scripts/glue_jobs/node_snapshots.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    output_frame = transformed_frame.resolveChoice(
+        [
+            ("id", "cast:bigint"),
+            ("benefactor_id", "cast:bigint"),
+            ("project_id", "cast:bigint"),
+            ("parent_id", "cast:bigint"),
+            ("file_handle_id", "cast:bigint")
+        ]
+    )
+
```
Comment (SandhraSokhal) [inline src/scripts/glue_jobs/node_snapshots.py:97]: skip writing if we have empty frame

File: src/scripts/glue_jobs/node_snapshots.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        return input_string[len('syn'):]
+        
+    return input_string
+
+# process the access record
+def transform(dynamic_record):
+    date = datetime.utcfromtimestamp(dynamic_record["change_timestamp"] / 1000.0)
+    
+    # This is the partition date
+    dynamic_record["change_date"] = date.strftime("%Y-%m-%d")
```
Comment (SandhraSokhal) [inline src/scripts/glue_jobs/node_snapshots.py]: partition should be snapshot_date

**Review by SandhraSokhal — APPROVED** [review]

**Pull request merged**